packages/utils/src/dal/mikro-orm/mikro-orm-repository.ts

```diff
@@ -116,6 +120,17 @@ export class MikroOrmBaseRepository<T extends object = object>
     throw new Error("Method not implemented.")
   }
 
+  upsertWithReplace(
+    data: unknown[],
+    upsertConfig: UpsertWithReplaceConfig<T> = {
+      relationsToUpsert: [],
+      relationsToSkip: [],
```

Comment:
do we need the `relationsToSkip` if `relationsToUpsert` is essentially a whitelist?

Reply:
+1

If not, I would change it to `include: []`

Reply:
@riqwan you might want to let the ORM take care of a relation (eg. if you pass `{collection: {id: "1234"}}` the ORM will link that to the product, so you don't want to either update nor skip it).

I think we can rethink what is being configured, but this works for now so I'll leave it as-is for the PR (and leave a comment in the code)

Reply:
@sradevski, can I get you show this with an example? I am not sure I understand.

Reply:
Sure, For example for a product:
```
{
    id: "prod1",
    tags: [{id: "tag1", value: "1"}, {id: "tag2", value: "2"}],
    collection: {id: "col1", title: "Col" },
}
```

And then we do an update to the tags

```
product.upsertWithReplace(
    [{id: "prod1", tags: [{id: "tag3", value: "3"}], collection: {id: "col2"}}],
    {relationsToUpsert: ["tags"], relationsToSkip: []}
)
```

What we expect here is that we upsert the tags, but we only reassign a new collection to the product ("col2"). We don't want to ignore that relation, nor upsert it, we simply want to let the ORM associate it to the product

@olivermrbl @riqwan does that make sense?

Reply:
But can't we handle this with just the `include`? 

Upsert tags on product and associate collection with product:
```
product.upsertWithReplace(
    [{id: "prod1", tags: [{id: "tag3", value: "3"}], collection: {id: "col2"}}],
    {include: ["tags"]}
)
```

Upsert tags on product and upsert collection:
```
product.upsertWithReplace(
    [{id: "prod1", tags: [{id: "tag3", value: "3"}], collection: {title: "new one"}}],
    {include: ["tags", "collection"]}
)
```

When is the skip needed?

Reply:
@olivermrbl maybe my example was a bit incomplete 😄 , Now imagine you have `variants` list in the data, but you don't want the function to do anything about it, so you want to tell `upsertWithReplace` to ignore the variants (and with just `include`, you can't do that.